### ISSUE TITLE
PSR4 based service discovery bug on Windows

### ISSUE THREAD

**Issue by podhy** [symfony/symfony#23103]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3.2

Hello,
I found bug on Windows when using PSR4 base service discovery.

When I've got in `app/config/services.yml`:
```yaml
services:
  _defaults:
      autowire: true
      autoconfigure: true

  AppBundle\:
      resource: ../../src/AppBundle/**/*Controller.php
```

All controllers (and not only controllers) are not discovered because of backslashes. This bug is only reproducible on Windows so all tests pass.

I've found little tricky solution to change lines 136-140 in `vendor/symfony/symfony/src/Symfony/Component/Config/Resource/GlobResource.php` from:
```php
foreach ($finder->followLinks()->sortByName()->in($this->prefix) as $path => $info) {
    if (preg_match($regex, substr($path, $prefixLen)) && $info->isFile()) {
        yield $path => $info;
    }
}
```

To:
```php
foreach ($finder->followLinks()->sortByName()->in($this->prefix) as $path => $info) {
    $normalizedPath = str_replace(array('/', '\\'), '/', $path); // normalize directory separator to UNIX slashes
    if (preg_match($regex, substr($normalizedPath, $prefixLen)) && $info->isFile()) {
        yield $path => $info;
    }
}
```

**Comment by bogdaniel** [comment]
Not sure if it is related to this bug i think this is the one .
`λ php bin\console cache:clear -e dev

Fatal error: Uncaught Symfony\Component\DependencyInjection\Exception\InvalidArgumentException: Expected to find class "AppBundle\Tests\Controller\MediaFolderControllerTest" in file "E:\Zenchron\zcms\src\AppBundle/Tests\Controller\MediaFolderControllerTest.php" while importing services from resource "../../src/AppBundle/*", but it was not found! Check the namespace prefix used with the resource. in E:\Zenchron\zcms\vendor\symfony\symfony\src\Symfony\Component\DependencyInjection\Loader\FileLoader.php:133
Stack trace:
#0 E:\Zenchron\zcms\vendor\symfony\symfony\src\Symfony\Component\DependencyInjection\Loader\FileLoader.php(61): Symfony\Component\DependencyInjection\Loader\FileLoader->findClasses('AppBundle\\', '../../src/AppBu...', '../../src/AppBu...')
#1 E:\Zenchron\zcms\vendor\symfony\symfony\src\Symfony\Component\DependencyInjection\Loader\YamlFileLoader.php(532): Symfony\Component\DependencyInjection\Loader\FileLoader->registerClasses(Object(Symfony\Component\DependencyInjection\Definition), 'AppBundle\\', '../../src/ in E:\Zenchron\zcms\vendor\symfony\symfony\src\Symfony\Component\Config\Loader\FileLoader.php on line 179
`

The file is there it exists and was created during the doctrine crud generation yet it can't find it.

**Comment by nicolas-grekas** [comment]
Your fix was good, see #23168
